TWLight/users/views.py

```diff
@@ -572,3 +573,108 @@ def get(self, request, pk, version, format=None):
+            raise Http404
+
+    def get_context_data(self, **kwargs):
+        context = super(CollectionUserView, self).get_context_data(**kwargs)
+        editor = self.get_object()
+        today = datetime.date.today()
+        proxy_bundle_authorizations = Authorization.objects.filter(Q(authorized_user=editor.user),
+                                                                   Q(date_expires__gte=today) |
+                                                                   Q(date_expires=None),
+                                                                   Q(partner__authorization_method=Partner.PROXY) |
+                                                                   Q(partner__authorization_method=Partner.BUNDLE)
+                                                                   ).order_by('partner')
+        proxy_bundle_authorizations_expired = Authorization.objects.filter(Q(authorized_user=editor.user),
+                                                                        Q(date_expires__lt=today),
+                                                                        Q(partner__authorization_method=Partner.PROXY) |
+                                                                        Q(partner__authorization_method=Partner.BUNDLE)
+                                                                           ).order_by('partner')
+        manual_authorizations = Authorization.objects.filter(Q(authorized_user=editor.user),
+                                                             Q(date_expires__gte=today) |
+                                                             Q(date_expires=None),
+                                                             Q(partner__authorization_method=Partner.EMAIL) |
+                                                             Q(partner__authorization_method=Partner.CODES) |
+                                                             Q(partner__authorization_method=Partner.LINK)
+                                                            ).order_by('partner')
+        manual_authorizations_expired = Authorization.objects.filter(Q(authorized_user=editor.user),
+                                                             Q(date_expires__lt=today),
+                                                             Q(partner__authorization_method=Partner.EMAIL) |
+                                                             Q(partner__authorization_method=Partner.CODES) |
+                                                             Q(partner__authorization_method=Partner.LINK)
+                                                            ).order_by('partner')
+        for each_authorization in proxy_bundle_authorizations:
+            if each_authorization.date_expires is not None:
+                if each_authorization.date_expires - today < timedelta(days=30):
+                    each_authorization.about_to_expire = True
+                    try:
```

Comment:
i.e. `Authorization.get_latest_app()`

Reply:
I can't find where else we're doing this but I distinctly remember needing to do either this or the reverse (get auth for an application) somewhere else.

Reply:
Ah, I might have been thinking of the reverse at https://github.com/WikipediaLibrary/TWLight/blob/master/TWLight/applications/views.py#L1027

Either way, this is a useful thing to have centralised, especially since you're doing it a few times in this file alone.